### ISSUE TITLE
Sort doesn't work on the first click

### ISSUE THREAD

**Issue by maxisam** [swimlane/angular2-data-table#254]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here
```

**Current behavior**
when click on column header at the very first time, the data in the table kinda moving around but not sort correctly.

after first click, everything works properly. (or not, sometime the first row is not really the first one like it should be.)

**Expected behavior**

should work at the first time

**Reproduction of the problem**

on demo page

**Please tell us about your environment:**
Win10/Chrome

* **Table version: latest
* **Angular version:** 2.0.x

Thank you for your great work !

**Comment by colinfang** [comment]
Thanks for the spot, I knew there is something weird with the sort. I had thought it was just slow responsiveness.

**Comment by amcdnl** [comment]
Fixed in latest release.

**Comment by maxisam** [comment]
ah.... it is kinda broken again. Check it on [demo](http://swimlane.github.io/angular2-data-table/#) page 

When you click it, it doesn't sort correctly. The first one is off.

**Comment by maxisam** [comment]
Another interesting behavior is, if you try it on filter demo,  sort it first and type something in the filter and the sorting will flip back to correct order.

**Comment by amcdnl** [comment]
Might be related to https://github.com/swimlane/angular2-data-table/blob/master/src/components/datatable.component.ts#L96 ...

**Comment by maxisam** [comment]
yeah, I was looking at that line too. But I think it is correct, because it flips back to correct order, which indicates the sorting function works correctly.

**Comment by scottmizo** [comment]
@amcdnl I think you are correct.  It appears that the setter is causing sortRows to be called a second time when the column header cell is clicked.

**Comment by amcdnl** [comment]
Fixed in 2.0
